src/ProjectFileIO.cpp

```diff
@@ -332,6 +332,9 @@ void ProjectFileIO::CheckpointThread()
          // Reset
          mCheckpointCurrentPages = 0;
       }
+      else
```

Comment:
Why?  What is sqlite going to do with a null pointer?  I could see the point if it returned a valid pointer and a non-zero return code.

Reply:
Indeed, the first thing it does is check for nullpointer then then returns:

static int sqlite3Close(sqlite3 *db, int forceZombie){
  if( !db ){
    /* EVIDENCE-OF: R-63257-11740 Calling sqlite3_close() or
    ** sqlite3_close_v2() with a NULL pointer argument is a harmless no-op. */
    return SQLITE_OK;
  }

Reply:
But... we don't know at that point that db is null.  We know that sqlite3_open returned a failure error code.

This page https://www.sqlite.org/c3ref/open.html says:

"Whether or not an error occurs when it is opened, resources associated with the database connection handle should be released by passing it to sqlite3_close() when it is no longer required."

So, a bad rc and a null pointer -- means nothing happens in that call, but there might be a bad rc and a non-null pointer -- which means a memory leak might happen without the call.